wrphttp/handler_test.go

```diff
@@ -213,7 +215,12 @@ func testWRPHandlerDecodeError(t *testing.T) {
 		errorEncoder       = func(actualCtx context.Context, actualErr error, _ http.ResponseWriter) {
 			errorEncoderCalled = true
 			assert.Equal(expectedCtx, actualCtx)
-			assert.Equal(expectedErr, actualErr)
+			assert.ErrorIs(actualErr, expectedErr,
+				fmt.Errorf("error [%v] doesn't contain error [%v] in its err chain",
+					actualErr, expectedErr))
+			if err, ok := actualErr.(httpError); assert.True(ok) {
```

Comment:
Can we use `assert.ErrrorsAs()` here? something like:

```go
var statusCoder gokithttp.StatusCoder
if assert.ErrorsAs(actualErr, statusCoder) {
    assert.Equal(expectedHTTPStatusCode, statusCoder.StatusCode())
}
```
StatusCoder interface: https://pkg.go.dev/github.com/go-kit/kit/transport/http#StatusCoder